### PR TITLE
fallback on `@Retry.maxDuration` exceeded, fixes #39

### LINKED ISSUE #39: @Retry with timeout and @Fallback throws TimeoutException when execution time exceeded maxDuration
See also https://issues.jboss.org/browse/THORN-2196

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

File: implementation/src/main/java/io/smallrye/faulttolerance/SimpleCommand.java [hunk 1]
```diff
@@ -85,15 +96,38 @@ protected Object run() throws Exception {
...
+        if (retryContext == null || !retryContext.shouldRetryOn(failure)) {
+            return fallback.get();
+        } else {
             return super.getFallback();
         }
-        return fallback.get();
+    }
+
+    private Throwable translateException() {
+        Exception e = executionResult.getExecutionException();
```
Comment (michalszynkiewicz) [inline implementation/src/main/java/io/smallrye/faulttolerance/SimpleCommand.java]: this isn't very good.

File: testsuite/basic/src/test/java/io/smallrye/faulttolerance/bulkhead/BulkheadTest.java [hunk 2]
```diff
@@ -50,22 +53,24 @@ public static JavaArchive createTestArchive() {
...
         int loop = QUEUE_SIZE * 2;
-        CountDownLatch startLatch = new CountDownLatch(loop);
+        CountDownLatch startLatch = new CountDownLatch(QUEUE_SIZE);
         CountDownLatch endLatch = new CountDownLatch(1);
         List<Future<String>> futures = new ArrayList<>();
         for (int i = 0; i < loop; i++) {
             futures.add(pingService.ping(startLatch, endLatch));
         }
-        startLatch.await(500, TimeUnit.MILLISECONDS);
+        assertTrue(startLatch.await(1000L, TimeUnit.MILLISECONDS));
```
Comment (mkouba) [inline testsuite/basic/src/test/java/io/smallrye/faulttolerance/bulkhead/BulkheadTest.java]: I think we should probably use something like this instead:
```java
startLatch.await(500, TimeUnit.MILLISECONDS);
Thread.sleep(500);
``` 
The reason is that `startLatch.await()` can eventually return before the waiting queue is full.

File: implementation/src/main/java/io/smallrye/faulttolerance/SimpleCommand.java (hunk 1 above)
Comment (mkouba) [reply implementation/src/main/java/io/smallrye/faulttolerance/SimpleCommand.java]: Then we should probably leave a comment and create an issue...

Comment (michalszynkiewicz) [reply implementation/src/main/java/io/smallrye/faulttolerance/SimpleCommand.java]: done

File: testsuite/basic/src/test/java/io/smallrye/faulttolerance/bulkhead/BulkheadTest.java (hunk 2 above)
Comment (michalszynkiewicz) [reply testsuite/basic/src/test/java/io/smallrye/faulttolerance/bulkhead/BulkheadTest.java]: done

**Pull request merged**